metpy/gridding/tests/test_interpolation.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2008-2015 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import division
+
+from metpy.gridding.interpolation import (barnes_weights, nn_point, cressman_weights,
+                                          cressman_point, barnes_point, natural_neighbor,
+                                          inverse_distance)
+
+from metpy.gridding.triangles import find_natural_neighbors, dist_2
+from metpy.gridding.gridding_functions import calc_kappa
+from metpy.cbook import get_test_data
+from scipy.spatial import cKDTree, Delaunay
+
+import numpy as np
+
+from numpy.testing import assert_array_almost_equal
+from numpy.testing import assert_almost_equal
+
+from metpy.gridding.points import generate_grid
+
+from scipy.spatial.distance import cdist
+
+
+def test_natural_neighbor():
+    r"""Tests natural neighbor interpolation function"""
+
+    xp = np.array([8, 67, 79, 10, 52, 53, 98, 34, 15, 58])
+    yp = np.array([24, 87, 48, 94, 98, 66, 14, 24, 60, 16])
```

Comment:
I think you should use another fixture here for the coords. You might consider using one for the grid (`xg` and `yg`) as well. The coords one (since there are like 5 places in here using them) would probably solve your codacy problem as well.